File: test/language/expressions/python/FullEnumerationTypeTest.py
import Expressions

class FullEnumerationTypeTest(Expressions.TestCase):
    def testBitSizeOfWithOptional(self):
        fullEnumerationTypeExpression = self.api.FullEnumerationTypeExpression(self.api.Color.RED, True)
        self.assertEqual(self.FULL_ENUMERATION_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL,
                         fullEnumerationTypeExpression.bitsizeof())

    def testBitSizeOfWithoutOptional(self):
        fullEnumerationTypeExpression = self.api.FullEnumerationTypeExpression()
        fullEnumerationTypeExpression.color = self.api.Color.BLUE
        self.assertEqual(self.FULL_ENUMERATION_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL,
                         fullEnumerationTypeExpression.bitsizeof())

    FULL_ENUMERATION_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL = 9
    FULL_ENUMERATION_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL = 8

File: test/language/expressions/cpp/FullEnumerationTypeTest.cpp
#include "gtest/gtest.h"

#include "expressions/full_enumeration_type/FullEnumerationTypeExpression.h"

namespace expressions
{
namespace full_enumeration_type
{

TEST(FullEnumerationTypeTest, bitSizeOfWithOptional)
{
    FullEnumerationTypeExpression fullEnumerationTypeExpression;
    fullEnumerationTypeExpression.setColor(Color::RED);
    fullEnumerationTypeExpression.setIsColorRed(true);

    const size_t fullEnumerationTypeExpressionBitSizeWithOptional = 9;
    ASSERT_EQ(fullEnumerationTypeExpressionBitSizeWithOptional, fullEnumerationTypeExpression.bitSizeOf());
}

TEST(FullEnumerationTypeTest, bitSizeOfWithoutOptional)
{
    FullEnumerationTypeExpression fullEnumerationTypeExpression;
    fullEnumerationTypeExpression.setColor(Color::BLUE);

    const size_t fullEnumerationTypeExpressionBitSizeWithoutOptional = 8;
    ASSERT_EQ(fullEnumerationTypeExpressionBitSizeWithoutOptional, fullEnumerationTypeExpression.bitSizeOf());
}

} // namespace full_enumeration_type
} // namespace expressions
